node/cli/src/service.rs

```diff
@@ -110,21 +112,20 @@ construct_service_factory! {
 					service.spawn_task(Box::new(select));
 				}
 
-				let grandpa_key = if service.config.disable_grandpa {
+				let grandpa_session_store = if service.config.disable_grandpa {
 					None
 				} else {
-					service.authority_key::<grandpa_primitives::AuthorityPair>()
+					service.get_session_store::<grandpa_primitives::AuthorityPair>()
 				};
 
 				let config = grandpa::Config {
-					local_key: grandpa_key.map(Arc::new),
 					// FIXME #1578 make this available through chainspec
 					gossip_duration: Duration::from_millis(333),
 					justification_period: 4096,
 					name: Some(service.config.name.clone())
 				};
 
-				match config.local_key {
+				match grandpa_session_store {
```

Comment:
`grandpa_session_store` is only `None` when `service.config.disable_grandpa` is true. In addition `grandpa_session_store` is only used within the `Some` branch of this `match`. How about doing `match service.config.disable_grandpa` here and constructing `grandpa_session_store` inside the `Some` branch?

Reply:
this is because the keystore is optional, we can only construct a session_store if we have a keystore.

Reply:
@tomaka made it optional recently to compile substrate to wasm

Reply:
Ah, right `service.get_session_store::<grandpa_primitives::AuthorityPair>()` also returns an `Option` which could be `None`. Thanks.